snap7/util.py

```diff
@@ -410,6 +410,44 @@ def get_real(bytearray_: bytearray, byte_index: int) -> float:
     real = struct.unpack('>f', struct.pack('4B', *x))[0]
     return real
 
+def set_fstring(bytearray_: bytearray, byte_index: int, value: str, max_length: int):
+    """Set space-padded fixed-length string value
+
+    Args:
+        bytearray_: buffer to write to.
+        byte_index: byte index to start writing from.
+        value: string to write.
+        max_length: maximum string length, i.e. the fixed size of the string.
+
+    Raises:
+        :obj:`TypeError`: if the `value` is not a :obj:`str`.
+        :obj:`ValueError`: if the length of the `value` is larger than the `max_size`
+        or 'value' contains non-ascii characters.
+
+    Examples:
+        >>> data = bytearray(20)
+        >>> snap7.util.set_fstring(data, 0, "hello world", 15)
+        >>> data
+            bytearray(b'hello world    \x00\x00\x00\x00\x00')
+    """
+    if not isinstance(value, str):
+        raise TypeError(f"Value value:{value} is not from Type string")
```

Comment:
to be consistent, I think validating the type should be handled by a type checker like mypy, and not be done manually.